### PR TITLE
Update for preshared secret

### PR DESCRIPTION
**Pull request by ThibaudDauce**

### CODE REVIEW

File: titanic.ipynb [hunk 1]
```diff
@@ -69,9 +66,10 @@
     "---\n",
     "## Step 2: Register (All participants)\n",
     "\n",
-    "To register, participants need a public PGP key. To generate PGP key pair, participants can use free software like [GPG](https://gnupg.org/download/index.html). In this example, public PGP were generated offline and stored in the folder `public_keys`.\n",
+    "To register, participants need a crypto context. To start a crypto context, all participants need to provide the shared words generated by the computation owner. \n",
     "\n",
-    "Note: At this time, participants who have several roles need distinct public keys for each. In our case, Alice will have to provide four different public keys."
+    "Note: The `CryptoContext` will generate all the necessary cryptographic keys (an asymmetric key pair and a symmetric key), you can override these keys.\n",
```
Comment (ccorsin) [inline titanic.ipynb]: "but you can also override these keys"?

File: titanic.ipynb [hunk 2]
```diff
@@ -163,13 +159,19 @@
    },
    "outputs": [],
    "source": [
-    "# Wait for the generation of the enclave identity\n",
+    "# Wait for the generation of the enclave identity, we are using the code provider object to do the waiting but \n",
```
Comment (ccorsin) [inline titanic.ipynb]: Could be interesting to precise that they can access stdout and stderr ? Or too much for demo, just in the doc ?

File: titanic.ipynb [hunk 3]
```diff
@@ -218,7 +211,7 @@
     "---\n",
     "## Step 6: Send data (Data Providers)\n",
     "\n",
-    "As for the code, Alice encrypts her data with a symmetric key before sending them."
+    "As for the code, Alice push her data, encrypted with the CryptoContext's symmetric key before sending."
```
Comment (ccorsin) [inline titanic.ipynb]: Alice pushes

File: titanic.ipynb [hunk 4]
```diff
@@ -353,14 +334,8 @@
...
    "source": [
-    "# Import package\n",
-    "from cosmian_secure_computation_client.crypto.helper import decrypt\n",
-    "\n",
     "# Download the result\n",
-    "encrypted_results = result_consumer1.fetch_results(computation.uuid)\n",
-    "\n",
-    "# Decrypt the result\n",
-    "result = decrypt(encrypted_results, symmetric_key_ro1)\n",
+    "result = result_consumer1.fetch_results(computation.uuid)\n",
```
Comment (ccorsin) [inline titanic.ipynb]: Why removed decryption ?

File: titanic.ipynb (hunk 2 above)
Comment (ThibaudDauce) [reply titanic.ipynb]: @seiteta What do you think of this?

File: titanic.ipynb [hunk 5]
```diff
@@ -69,9 +66,10 @@
     "---\n",
     "## Step 2: Register (All participants)\n",
     "\n",
-    "To register, participants need a public PGP key. To generate PGP key pair, participants can use free software like [GPG](https://gnupg.org/download/index.html). In this example, public PGP were generated offline and stored in the folder `public_keys`.\n",
+    "To register, participants need a crypto context. To start a crypto context, all participants need to provide the shared words generated by the computation owner. \n",
     "\n",
-    "Note: At this time, participants who have several roles need distinct public keys for each. In our case, Alice will have to provide four different public keys."
+    "Note: The `CryptoContext` will generate all the necessary cryptographic keys (an asymmetric key pair and a symmetric key), but you can also override these keys.\n",
+    "Note: If you split the the code in multiple files, you need to save and load the same crypto context every time."
```
Comment (ThibsG) [inline titanic.ipynb]:
```suggestion
    "Note: If you split the code in multiple files, you need to save and load the same crypto context every time."
```

File: titanic.ipynb (hunk 2 above)
Comment (ThibsG) [reply titanic.ipynb]: 1 line about this doesn't seem too much to me

File: titanic.ipynb [hunk 6]
```diff
@@ -137,22 +136,19 @@
...
+    "code_provider.upload(computation_uuid, path)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "---\n",
-    "## Step 4: Approve participants (Computation Owner)\n",
+    "## Step 4: Wait the identity of the enclave\n",
```
Comment (ThibsG) [inline titanic.ipynb]:
```suggestion
    "## Step 4: Wait for the identity of the enclave\n",
```

**Review by ThibsG — APPROVED** [review]
This notebook was pretty outdated anyway, and with last commit we get everything we want to be up to date

**Review by ccorsin — APPROVED** [review]

**Pull request merged**